src/s11/handlers/create_session_handler.c

```diff
@@ -77,6 +77,26 @@ bswap8_array(uint8_t *src, uint8_t *dest, uint32_t len)
 	return;
 }
 
+/**
+ Transform the imsi format for s11 interface packet
+*/
+void
+format_imsi_for_s11(uint8_t *src, uint8_t *dest, uint32_t len)
+{
+        // Change it to a human readable number
+        uint8_t imsi_tmp[BINARY_IMSI_LEN];
+        bswap8_array(src, imsi_tmp, BINARY_IMSI_LEN);
+
+        // Remove the extra 4 bits and fill in 1 according to the spec
+        for (uint32_t i = 0; i < len - 1; i++){
+                dest[i] = ( ( imsi_tmp[i] & 0x0F ) << 4 ) | ( ( imsi_tmp[i + 1] & 0xF0 ) >> 4 );
+        }
+        dest[len - 1] = ( ( imsi_tmp[len - 1] & 0x0f ) << 4 ) | ( 0xF0 >> 4) ;
```

Comment:
dest[len - 1] = ( ( imsi_tmp[len - 1] & 0x0f ) << 4 ) | ( 0xF0 >> 4) ;
This can be = ( ( imsi_tmp[len - 1] & 0x0f ) << 4 ) | ( 0x0F) ; But will accept above code as that is consistent with rest of the bitswaps code lines and hence easy to understand.

Reply:
Thank you!

Reply:
Next time, after you agree to merge, please also leave a comment "+1" here. After each patch got 2 score , then the patch can be merged.